testsupport/spaceprovisionerconfig/spaceprovisionerconfig.go

```diff
@@ -54,12 +57,16 @@ func UpdateForCluster(t *testing.T, await *wait.Awaitility, referencedClusterNam
 			require.Fail(t, err.Error())
 		}
 
-		// make the originalSpc look like we freshly obtained it from the server and updated its fields
-		// to look like the original.
-		originalSpc.Generation = currentSpc.Generation
-		originalSpc.ResourceVersion = currentSpc.ResourceVersion
+		_, err = await.UpdateSpaceProvisionerConfig(t, originalSpc.Name,
+			func(spcfg *toolchainv1alpha1.SpaceProvisionerConfig) {
+				// make the originalSpc look like we freshly obtained it from the server and updated its fields
+				// to look like the original.
+				spcfg.Generation = currentSpc.Generation
+				spcfg.ResourceVersion = currentSpc.ResourceVersion
```

Comment:
there is no need for this hack, the loop ensures that it gets the fresh object
```suggestion
```

Reply:
My bad!! Thanks for noticing it!